score-lang-cli/src/main/java/org/openscore/lang/cli/services/ScoreServicesImpl.java

```diff
@@ -131,12 +143,43 @@ public synchronized void onEvent(ScoreEvent scoreEvent) throws InterruptedExcept
                         printWithColor(Ansi.Color.YELLOW, prefix + taskName);
                     }
                     break;
+                case ScoreLangConstants.EVENT_OUTPUT_END:
+                    if(quietListener){
+                        printFlowOutputs(data);
+                    }
+                    break;
                 case EVENT_EXECUTION_FINISHED :
+                   if (!flowFinished.get()){
+                        flowFinished.set(true);
+                    }
                     printFinishEvent(data);
                     break;
             }
         }
 
+        private void printFlowOutputs(Map<String, Serializable> data) {
```

Comment:
this gets the outputs of the flow, truncates any value to length 30 , puts "(empty)" in empty values and replaces newlines with space. it isn't so complex, what do you think I could simplify/remove?